tests/test_new_command_system.py

```diff
@@ -0,0 +1,204 @@
+        "description": "Hello!",
+        "description_localizations": {},
+        "dm_permission": True,
+        "name": "command",
+        "name_localizations": {},
+        "options": [
+            {
+                "description": "hi",
+                "name": "option",
+                "required": True,
+                "type": interactions.OptionType.STRING,
+            }
+        ],
+        "type": 1,
+    }
+
+
+def test_basic_command_with_not_required_option_with_read_from_kwargs(fake_client):
+    @fake_client.command()
+    @interactions.option("hi")
+    async def command(ctx, option: str = ""):
+        """Hello!"""
+
+    assert command.full_data == {
+        "default_member_permissions": "2147483648",
+        "description": "Hello!",
+        "description_localizations": {},
+        "dm_permission": True,
+        "name": "command",
+        "name_localizations": {},
+        "options": [
+            {
+                "description": "hi",
+                "name": "option",
+                "required": False,
+                "type": interactions.OptionType.STRING,
+            }
+        ],
+        "type": 1,
+    }
+
+
+def test_basic_command_with_required_option_with_read_from_option_decorator(fake_client):
+    @fake_client.command()
+    @interactions.option("hi", type=interactions.OptionType.USER, required=True)
+    async def command(ctx, option: str = None):
+        """Hello!"""
+
+    assert command.full_data == {
+        "default_member_permissions": "2147483648",
+        "description": "Hello!",
+        "description_localizations": {},
+        "dm_permission": True,
+        "name": "command",
+        "name_localizations": {},
+        "options": [
+            {
+                "description": "hi",
+                "name": "option",
+                "required": True,
+                "type": interactions.OptionType.USER,
+            }
+        ],
+        "type": 1,
+    }
+
+
+def test_basic_command_with_not_required_option_with_read_from_option_decorator(fake_client):
+    @fake_client.command()
+    @interactions.option("hi", type=interactions.OptionType.USER, required=False)
+    async def command(ctx, option: str):
+        """Hello!"""
+
+    assert command.full_data == {
+        "default_member_permissions": "2147483648",
+        "description": "Hello!",
+        "description_localizations": {},
+        "dm_permission": True,
+        "name": "command",
+        "name_localizations": {},
+        "options": [
+            {
+                "description": "hi",
+                "name": "option",
+                "required": False,
+                "type": interactions.OptionType.USER,
+            }
+        ],
+        "type": 1,
+    }
+
+
+def test_basic_command_with_connector_option(fake_client):
+    @fake_client.command()
+    @interactions.option(
+        "hi", name="hi", type=interactions.OptionType.USER, required=True, connector="d"
```

Comment:
```suggestion
        "hi", name="hi", type=interactions.OptionType.USER, required=True
```
Connector is automatically set as parameter name if name= is specified

Reply:
Oh cool then I have to make two test cases for it xD

Reply:
Oh yeah btw required= is automatic too based on whether there is a default for the parameter

Reply:
If you looked at the function names you'd know 😅

I'm checking if the overriding works

If any of the checks fails then something is not ok

Reply:
> Connector is automatically set as parameter name if name= is specified

uhh.. yeah so about that:
```py
============================================================================================================== FAILURES =============================================================================================================== 
___________________________________________________________________________________ test_basic_command_with_connector_option_with_read_from_kwargs ____________________________________________________________________________________ 

fake_client = <interactions.client.bot.Client object at 0x0000022169ADECE0>

    def test_basic_command_with_connector_option_with_read_from_kwargs(fake_client):
        @fake_client.command()
        @interactions.option("hi", name="hi")
        async def command(ctx, d: interactions.Member):
            """Hello!"""

>       assert command.full_data == dict(
            default_member_permissions="2147483648",
            description="Hello!",
            description_localizations={},
            dm_permission=True,
            name="command",
            name_localizations={},
            options=[
                {
                    "connector": "d",
                    "description": "hi",
                    "name": "hi",
                    "required": True,
                    "type": interactions.OptionType.USER,
                }
            ],
            type=1,
        )
E       AssertionError: assert {'default_mem...n': True, ...} == {'default_mem...n': True, ...}
E         Omitting 7 identical items, use -vv to show
E         Differing items:
E         {'options': [{'description': 'hi', 'name': 'hi', 'required': True, 'type': <OptionType.USER: 6>}]} != {'options': [{'connector': 'd', 'description': 'hi', 'name': 'hi', 'required': True, ...}]}
E         Use -v to get more diff

tests\test_new_command_system.py:131: AssertionError
======================================================================================================= short test summary info ======================================================================================================= 
FAILED tests/test_new_command_system.py::test_basic_command_with_connector_option_with_read_from_kwargs - AssertionError: assert {'default_mem...n': True, ...} == {'default_mem...n': True, ...}
```

Reply:
oh I see how you did it, this is interesting

Reply:
Wait what happened

Reply:
Well the "connector" is shown in the option dict (check code above) if its put inside the option decorator but not if its made automatically and I thought that was a bug but command.connectors still is correct